src/modules/ipc.cpp

```diff
@@ -59,11 +60,33 @@ namespace modules {
     for (auto& hook : m_hooks) {
       hook->command = pid_token(hook->command);
     }
-    m_formatter->add(DEFAULT_FORMAT, TAG_OUTPUT, {TAG_OUTPUT});
 
+    m_formatter->add(DEFAULT_FORMAT, TAG_LABEL, {TAG_LABEL});
+    m_formatter->add_optional("output", {});
+
+    if (m_formatter->has_format("output")) {
```

Comment:
I think what you are looking for is `m_formatter->has("<output>", DEFAULT_FORMAT)`.

Reply:
I understood my mistake now. I took some methods used here to behave completely different. 
`cpp 
    if (m_formatter->has(TAG_OUTPUT)) {
      m_conf.warn_deprecated(name(), "<output>", "<label>");
      m_formatter->add(m_format, TAG_OUTPUT, {TAG_OUTPUT});

      for (size_t i = 0; i < m_hooks.size(); i++) {
        string format_i = "format-" + to_string(i);
        m_formatter->add_optional(format_i, {TAG_OUTPUT});
      }

    } else {
      if (m_formatter->has(TAG_LABEL)) {
        m_label = load_optional_label(m_conf, name(), TAG_LABEL, "%output%");

        for (size_t i = 0; i < m_hooks.size(); i++) {
          string format_i = "format-" + to_string(i);
          m_formatter->add_optional(format_i, {TAG_LABEL});
        }
      }
```
```cpp
  bool ipc_module::build(builder* builder, const string& tag) const {
    if (tag == TAG_LABEL) {
      builder->node(m_label);
      return true;
    } else if (tag == TAG_OUTPUT) {
      builder->node(m_output);
      return true;
    }
    return false;
```
However with this particular approach.I Should I implement in such a way that all format-x and default format should follow  either label or output and not a combination of both ? Or Should I allow use of both output and label by the user. Seems to me that we should only allow tag_label or tag_output. 
Also another edge case was that If I do format= <output> <label>, Should I set whether the user has to follow <output> or <label> based on the default format ?

Reply:
There is no reason to restrict the user from only using `<output>` or `<label>` but never both, that's just more work to check for that.

In `format-x` we don't need to support `<output>` at all, because it's a new option and thus can't break existing configs.